File: OpenLibraryBackend/src/main/java/com/open/library/POJO/Category.java
package com.open.library.POJO;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

@Entity
@Table(name = "categories")
@NoArgsConstructor
@AllArgsConstructor
@Data
@DynamicInsert
@DynamicUpdate
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "category_id")
    private Long id;

    private String name;

    @Column(name = "code", unique = true, nullable = false)
    private String code;

    private boolean is_activated;

    private boolean is_deleted;
}

File: OpenLibraryFrontend/src/app/shared/models/Category.ts
export class Category {

    id!: number;
    name!: string;
    code!: string;
    is_activated?: boolean;
    is_deleted?: boolean;
    
    constructor(id: number, name: string, code: string, is_activated: boolean, is_deleted: boolean) {
        this.id = id;
        this.name = name;
        this.code = code;
        this.is_activated = is_activated;
        this.is_deleted = is_deleted;
    }
}